packages/ckeditor5-engine/tests/model/utils/insertcontent.js

```diff
@@ -1299,6 +1299,358 @@ describe( 'DataController utils', () => {
+
+				expect( expectedMarker.getRange().start.path ).to.deep.equal( [ 0, 4 ] );
+				expect( expectedMarker.getRange().end.path ).to.deep.equal( [ 1, 1 ] );
+			} );
+
+			it( 'should create two markers that have intersection between multiple elements', () => {
+				setData( model, '<paragraph>Foo[]</paragraph>' );
+
+				insertHelper( '<paragraph>>bar</paragraph><paragraph><imageInline></imageInline>test</paragraph>', {
+					'comment-thread-1': { start: [ 0, 1 ], end: [ 1, 3 ] },
+					'comment-thread-2': { start: [ 0, 2 ], end: [ 1, 4 ] }
+				} );
+
+				const expectedFirstMarker = model.markers.get( 'comment-thread-1' );
+				const expectedSecondMarker = model.markers.get( 'comment-thread-2' );
+
+				expect( getData( model ) )
+					.to.equal( '<paragraph>Foo>bar</paragraph><paragraph><imageInline></imageInline>test[]</paragraph>' );
+
+				expect( expectedFirstMarker.getRange().start.path ).to.deep.equal( [ 0, 4 ] );
+				expect( expectedFirstMarker.getRange().end.path ).to.deep.equal( [ 1, 3 ] );
+				expect( expectedSecondMarker.getRange().start.path ).to.deep.equal( [ 0, 5 ] );
+				expect( expectedSecondMarker.getRange().end.path ).to.deep.equal( [ 1, 4 ] );
+			} );
+
+			it( 'should create marker on imageInline inside paragraph after inserting at the beginning', () => {
+				setData( model, '<paragraph>[]Foo</paragraph>' );
+
+				insertHelper( '<paragraph><imageInline></imageInline></paragraph><paragraph>>bar</paragraph>', {
+					'comment-thread-1': { start: [ 0, 0 ], end: [ 0, 1 ] }
+				} );
+
+				const expectedMarker = model.markers.get( 'comment-thread-1' );
+
+				expect( getData( model ) )
+					.to.equal( '<paragraph><imageInline></imageInline></paragraph><paragraph>>bar[]Foo</paragraph>' );
+
+				expect( expectedMarker.getRange().start.path ).to.deep.equal( [ 0, 0 ] );
+				expect( expectedMarker.getRange().end.path ).to.deep.equal( [ 0, 1 ] );
+			} );
+
+			it( 'should create marker after inserting imageInline and paragraph inside existing paragraph', () => {
+				setData( model, '<paragraph>F[]oo</paragraph>' );
+
+				insertHelper( '<imageInline></imageInline><paragraph>>bar</paragraph>', {
+					'comment-thread-1': { start: [ 0 ], end: [ 1, 2 ] }
+				} );
+
+				const expectedMarker = model.markers.get( 'comment-thread-1' );
+
+				expect( getData( model ) )
+					.to.equal( '<paragraph>F<imageInline></imageInline></paragraph><paragraph>>bar[]oo</paragraph>' );
+
+				expect( expectedMarker.getRange().start.path ).to.deep.equal( [ 0, 1 ] );
+				expect( expectedMarker.getRange().end.path ).to.deep.equal( [ 1, 2 ] );
+			} );
+
+			it( 'should create marker which starts in paragraph and ends inside the beginning of next paragraph', () => {
+				setData( model, '<paragraph>Foo[]</paragraph>' );
+
+				insertHelper( '<paragraph>>bar</paragraph><paragraph></paragraph>', {
+					'comment-thread-1': { start: [ 0, 1 ], end: [ 1, 0 ] }
+				} );
+
+				const expectedMarker = model.markers.get( 'comment-thread-1' );
```

Comment:
My intention wasn't to check getData output and only check markers positions, because I created similar tests where I checked getData. Anyways, I will update it.